include/odp/api/spec/event.h

```diff
@@ -143,6 +143,20 @@ void *odp_event_user_area(odp_event_t event);
  */
 void *odp_event_user_area_and_flag(odp_event_t event, int *flag);
 
+/**
+ * Set event user flag
+ *
+ * Set (or clear) the user flag.
+ *
+ * The user flag maps to the user flag value of underlying event type (e.g.
+ * odp_packet_user_flag() for packets). If the event does not have user flag,
+ * nothing is done.
+ *
+ * @param event   Event handle
+ * @param val     New value for the flag. Zero clears the flag, other values set the flag.
+ */
+void odp_event_user_flag_set(odp_event_t event, int val);
```

Comment:
We could add a (bool) return value that indicates if event type supports user flag